Enforce MaxPayload on the client

wallyqs opened the pull request:
Makes the client proactively follow the `MaxPayload` size which was announced by the server upon connect. After this change, the client would get an `ErrMaxPayload` error when trying to publish a message which is larger than this limit.

derekcollison commented:
Should it be settable by the client at all? Should simply be something that
the client can query but its set by the server and enforced by the client
upon send.

On Tue, Sep 1, 2015 at 8:15 AM, Waldemar Quevedo <notifications@github.com>
wrote:

> Exposes EnforceMaxPayload and MaxPayload options which can be set by the
> client to enable proactively avoiding to send messages which are over the
> limit which was specified by the server when connecting.
>
> By default, EnforceMaxPayload is deactivated and has to be explicitly
> enabled when customizing the connection. In case activated, the client
> would get an ErrMaxPayload error when publishing.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/nats-io/nats/pull/64
> Commit Summary
>
>    - Enable option for enforcing MaxPayload on the client.
>    - Use ephemeral port on max payload test.
>
> File Changes
>
>    - *M* conn_test.go
>    <https://github.com/nats-io/nats/pull/64/files#diff-0> (67)
>    - *M* nats.go <https://github.com/nats-io/nats/pull/64/files#diff-1>
>    (15)
>
> Patch Links:
>
>    - https://github.com/nats-io/nats/pull/64.patch
>    - https://github.com/nats-io/nats/pull/64.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nats-io/nats/pull/64>.
>

wallyqs commented:
Thanks for the feedback, I modified it so that the client follows the max_payload size that is sent from server in the INFO message and removed the `EnforceMaxPayload`, reworded the PR to reflect this.

derekcollison commented:
We should have a MaxPayload() method on the connection, that way its read
only.

On Tue, Sep 1, 2015 at 2:03 PM, Waldemar Quevedo <notifications@github.com>
wrote:

> Thanks for the feedback, I modified it so that the client follows the
> max_payload size that is sent from server in the INFO message and removed
> the EnforceMaxPayload, reworded the PR to reflect this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nats-io/nats/pull/64#issuecomment-136860894>.
>

wallyqs commented:
Thanks for the comments, I made some updates to the PR addressing the issues you mentioned in case you can take a look.
